server/config/checkstyle/checkstyle.xml

```diff
@@ -0,0 +1,225 @@
+    <module name="TypecastParenPad"/>
+    <module name="WhitespaceAfter"/>
+    <module name="WhitespaceAround"/>
+
+    <!-- Modifier Checks                                    -->
+    <!-- See https://checkstyle.org/config_modifiers.html -->
+    <module name="ModifierOrder"/>
+    <module name="RedundantModifier"/>
+
+    <!-- Checks for blocks. You know, those {}'s         -->
+    <!-- See https://checkstyle.org/config_blocks.html -->
+    <module name="AvoidNestedBlocks"/>
+    <module name="EmptyBlock"/>
+    <module name="LeftCurly"/>
+    <module name="NeedBraces"/>
+    <module name="RightCurly"/>
+
+    <!-- Checks for common coding problems               -->
+    <!-- See https://checkstyle.org/config_coding.html -->
+    <module name="EmptyStatement"/>
+    <module name="EqualsHashCode"/>
+    <module name="HiddenField"/>
+    <module name="IllegalInstantiation"/>
+    <module name="InnerAssignment"/>
+    <module name="MagicNumber"/>
+    <module name="MissingSwitchDefault"/>
+    <module name="MultipleVariableDeclarations"/>
+    <module name="SimplifyBooleanExpression"/>
+    <module name="SimplifyBooleanReturn"/>
+
+    <!-- Checks for class design                         -->
+    <!-- See https://checkstyle.org/config_design.html -->
+    <!-- This only really makes sense if we're building a reasonably
+         sophisticate OO library, and that kind of design really doesn't
+         come up in this class, because the server library (e.g., Javalin)
+         encapsulates all that design work, leaving us to just act around
+         the "leaves" of the design.
+    <module name="DesignForExtension"/>
+    -->
+    <module name="FinalClass"/>
+    <!-- If everything is static in a class, then it shouldn't have
+         an accessible constructor. This is both subtle and
+         confusing, and not likely to come up much in the class,
+         so we'll just leave it out.
+    <module name="HideUtilityClassConstructor"/>
+    -->
+    <module name="InterfaceIsType"/>
+    <module name="VisibilityModifier"/>
+
+    <!-- Miscellaneous other checks.                   -->
+    <!-- See https://checkstyle.org/config_misc.html -->
+    <module name="ArrayTypeStyle"/>
+    <!-- I (Nic) really likes the *intent* of this rule,
+         but it's really trying to use CheckStyle to fix
+         a mistake in the design of Java. This just confuses
+         students, and requires a ton of extra typing
+         that probably doesn't buy folks much.
+    <module name="FinalParameters"/>
+    -->
+    <module name="TodoComment"/>
+    <module name="UpperEll"/>
+
+    <!-- https://checkstyle.org/config_filters.html#SuppressionXpathFilter -->
+    <module name="SuppressionXpathFilter">
+      <property name="file" value="${org.checkstyle.sun.suppressionxpathfilter.config}"
+                default="checkstyle-xpath-suppressions.xml" />
+      <property name="optional" value="true"/>
+    </module>
+
+  </module>
+
+</module>
```

Comment:
I really like the set of rules you've chosen—these look good!

Reply:
I wish I could take more credit, but these are really just the "old" Sun rules, which I've generally found quite useful.